logstash-core/lib/logstash/runner.rb

```diff
@@ -148,25 +148,32 @@ def run(args)
     begin
       LogStash::SETTINGS.from_yaml(LogStash::SETTINGS.get("path.settings"))
     rescue => e
-      @logger.subscribe(STDOUT)
-      @logger.warn("Logstash has a new settings file which defines start up time settings. This file is typically located in $LS_HOME/config or /etc/logstash. If you installed Logstash through a package and are starting it manually please specify the location to this settings file by passing in \"--path.settings=/path/..\" in the command line options")
-      @logger.fatal("Failed to load settings file from \"path.settings\". Aborting...", "path.settings" => LogStash::SETTINGS.get("path.settings"), "exception" => e.class, "message" => e.message)
+      $stderr.puts "ERROR: Logstash has a new settings file which defines start up time settings. This file is typically located in $LS_HOME/config or /etc/logstash. If you installed Logstash through a package and are starting it manually please specify the location to this settings file by passing in \"--path.settings=/path/..\" in the command line options"
+      $stderr.puts "ERROR: Failed to load settings file from \"path.settings\". Aborting... path.setting=#{LogStash::SETTINGS.get("path.settings")}, exception=#{e.class}, message=>#{e.message}"
       exit(-1)
     end
 
+    # Configure Logstash logging facility, this need to be done before everything else to
+    # make sure the logger has the correct settings and the log level is correctly defined.
+    # TODO(talevy): cleanly support `path.logs` setting in log4j
+    log4j_config_location = setting("path.settings") + "/log4j2.properties"
+    unless java.lang.System.getProperty("log4j.configurationFile")
+      LogStash::Logging::Logger::initialize(log4j_config_location)
+    end
+
     super(*[args])
   end
 
   def execute
     require "logstash/util"
     require "logstash/util/java_version"
     require "stud/task"
-    require "cabin" # gem 'cabin'
-    require "logstash/logging/json"
 
-    # Configure Logstash logging facility, this need to be done before everything else to
-    # make sure the logger has the correct settings and the log level is correctly defined.
-    configure_logging(setting("path.log"), setting("log.level"))
+    LogStash::Logging::Logger::configure_logging(setting("log.level"))
```

Comment:
> @fbaligand : Either you define a custom log4j2 Lookup implementation where you inject logstash settings  
> @talevy : This would be a nice thing to do for initial loading of the configuration, but custom log4j configs that may not use / understand these lookups will not see log.level respected upon loading. I understand we can inform users in the docs, but it feels like a simple thing to ignore.

- firstly, I think that very few people will make a custom log4j configuration (maybe only me :p). I mean : define a totally different file, not just customizing some logger levels in provided log4j configuration.
- secondly, people that make custom log4j config would certainly take inspiration from provided `log4j2.properties`, see the `Lookup` references, and so, understand the idea. => I don't think it "requires" a documentation.
- thirdly, for `path.log` (or `path.logs`), at my knowledge, you have no other way than `Lookup` to do that
- at last, this is what elasticsearch uses for its log4j configuration

So up to me, go for Lookup ! :)